hardware/raspberrypi/booteeprom.md

```diff
@@ -7,38 +7,97 @@ The Raspberry Pi 4 has an SPI-attached EEPROM (4MBits/512KB), which contains cod
  - Raspberry Pi 4 bootup procedure and SDRAM setup is considerably more complicated than on the previous Raspberry Pi models, so there is more risk inherent in code that's permanently incorporated in the ROM of the SoC.
  - USB has moved to a PCIe bus, and the Gigabit Ethernet driver is completely different to previous models, so again, having it permanently fixed into the ROM of the SoC was not feasible.
  - A small SPI EEPROM allows bugs to be fixed and features to be added after launch, in the field.
- - The local modifiable state means that OTP bootmode settings will not be required for PXE or USB mass storage boot on the Raspberry Pi 4. There are no user-modifiable OTP bootmode bits on Pi 4.
+ - The local modifiable state means that OTP bootmode settings will not be required for network or USB mass storage boot on the Raspberry Pi 4. There are no user-modifiable OTP bootmode bits on Pi 4.
 
-## PXE and USB Boot
+## Network and USB boot
 
-Support for these additional bootmodes will be added in the future via optional bootloader updates. The current schedule is to release PXE boot first, then USB boot.
+Support for these additional bootmodes will be added in the future via optional bootloader updates. The current schedule is to release network boot first, then USB boot.
 
 ## Is the bootloader working correctly?
 
 To check that the bootloader is working correctly, turn off the power, unplug everything from the Raspberry Pi 4, including the SD card, and then turn the power back on. If the green LED blinks with a repeating pattern then the bootloader is running correctly, and indicating that `start*.elf` has not been found. Any other actions imply that the bootloader is not working correctly and should be reinstalled using `recovery.bin`.
 
-## recovery.bin
+## recovery.bin - rescue image
 
 If the EEPROM needs updating or has somehow become corrupted, it can be reflashed using a fresh SD card with a copy of `recovery.bin` in the first partition of an SD card, formatted to FAT32.
 
 `recovery.bin` is a special utility which runs directly from the SD card and updates the EEPROM — it is not in itself a bootloader. It flashes the green LED rapidly (forever) upon success. Because it's not a bootloader, it won't load `start*.elf`, so once you see the green LED flashing rapidly, just re-insert a regular Raspbian SD card and reboot the Pi.
 
 It can be downloaded from the [raspberrypi.org downloads page](https://www.raspberrypi.org/downloads/).
 
+## Updating the bootloader
+
+To update the bootloader for new features or bug fixes we recommend installing the rpi-eeprom Raspbian package. This installs a background service which runs at boot and checks if a critical update is available. If so, it schedules the update to be applied at the next reboot.
```

Comment:
What about "We recommend setting up your Pi so that it automatically updates the bootloader: this means you will get new features and bug fixes as they are released. Bootloader updates are performed by the `rpi-eeprom` package, which installs a service that runs at boot-time to check for critical updates."

Also, does it really need two reboots to get the new firmware, or is the check and install done on the same boot?

Reply:
That's a good point. I think the package is currently set up not to restart the service when it updates. If I remove that, it can schedule the update for next reboot whenever the package is updated with the new firmware. That would save a pointless reboot.

Reply:
I'd be happy with that change. I tested systemctl mask and it has the desired behaviour of preventing the service from running without having to install it first.

Reply:
@andrum99 that reads better I'll change the text

Reply:
Forgot to mention - my  suggestion uses backticks round rpi-eeprom, which is Markdown for inline code - see https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet

Reply:
updated with backticks ... not sure what's wrong with ascii docs ;-)